apps/regattatimer/app.js

```diff
@@ -0,0 +1,335 @@
+            ]
+          }
+        ]}, {lazy: true});
+
+        this.interval = setInterval((function() {
+          this.layout.daytime.label = require("locale").time(new Date(), 1);
+          this.layout.render();
+
+          // keeps the watch screen lit up
+          g.flip();
+        }).bind(this), 1000);
+    },
+    setLayoutStartMinSec: function() {
+      g.clear();
+
+      this.layout = new Layout({
+        type: "v",
+        bgCol: this.theme.bgColor,
+        c: [
+          {
+            type: "h",
+            c: [
+              {type: "txt", font: "Anton", label: "4", col: this.theme.fgColor, id: "minutes", fillx: 1, filly: 1},
+              {type: "txt", font: "Anton", label: "59", col: this.theme.fgColor, id: "seconds", fillx: 1, filly: 1},
+            ]
+          }
+        ]}, {lazy: true}
+      );
+    },
+    setLayoutStartSec: function() {
+      g.clear();
+
+      this.layout = new Layout({
+        type: "v",
+        bgCol: this.theme.bgColor,
+        c:[
+          {type: "txt", font: "Anton", label: "", fillx: true, filly: true, col: this.theme.fgColor, id: "seconds"},
+        ]}, {lazy: true});
+    },
+    setLayoutRace: function() {
+      g.clear();
+
+      this.layout = new Layout({
+        type: "v",
+        bgCol: this.theme.bgColor,
+        c: [
+          {type: "txt", font: "20%", label: "00:00:00", col: this.theme.fgColor, pad: 4, filly: 1, fillx: 1, id: "racetime"},
+          {type: "txt", font: "15%", label: "-", col: this.theme.fgColor, pad: 4, filly:1, fillx:1, id: "daytime"},
+          // horizontal
+          {type: "h", c: [
+            {type: "txt", font: "10%", label: this.translate("speed"), col: this.theme.fgColor, pad:4, fillx:1, filly:1},
+            {type: "txt", font: "20%", label: "0", col: this.theme.fgColor, pad:4, fillx:1, filly:1, id: "speed"},
+            {type: "txt", font: "10%", label: this.translate("speed_unit"), col: this.theme.fgColor, pad:4, fillx:1, filly:1},
+          ]},
+          {type: "h", c: [
+            {type:"img", pad: 2, src: this.theme["icons"].satellites()},
+            {type: "txt", font: "10%", label: "0", col: this.theme.fgColor, pad: 2, filly:1, id: "satellites"},
+            // hacky, use empty element with fillx to push the other elments to the left an right side
+            {type: undefined, pad: 2, fillx: 1},
+            {type:"img", pad: 2, src: this.theme["icons"].battery()},
+            {type: "txt", font: "10%", label: "-", col: this.theme.fgColor, pad: 2, filly: 1, id: "battery"},
+          ]}
+      ]}, {lazy: true});
+    }
+  };
+}
+
+var regattatimer = Regattatimer();
+regattatimer.init();
+
+if(regattatimer.settings.gps) {
+  Bangle.setGPSPower(1);
+  Bangle.on('GPS', regattatimer.onGPS.bind(regattatimer));
+}
+
+Bangle.on('kill', function() {
+  Bangle.setLCDPower(0);
+  Bangle.setLCDTimeout(10);
+
+  if(regattatimer.settings.gps) {
+    Bangle.setGPSPower(0);
+  }
+});
```

Comment:
This is usually not needed since the watch reboots on exit and loads its defaults.

Reply:
Always open for improvements. If it is the case the watch reboots on every app exit and GPS is turned off by default this is unnecessary.